bot/cogs/moderation/management.py

```diff
@@ -221,30 +220,41 @@ def infractions_cog(self) -> Infractions:
             max_size=1000
         )
 
-    def infraction_to_string(self, infraction: utils.Infraction) -> str:
+    async def infraction_to_string(self, infraction: utils.Infraction) -> str:
         """Convert the infraction object to a string representation."""
         actor_id = infraction["actor"]
         guild = self.bot.get_guild(constants.Guild.id)
         actor = guild.get_member(actor_id)
         active = infraction["active"]
         user_id = infraction["user"]
+        user = guild.get_member(user_id)
         hidden = infraction["hidden"]
         created = time.format_infraction(infraction["inserted_at"])
         if infraction["expires_at"] is None:
             expires = "*Permanent*"
         else:
             expires = time.format_infraction(infraction["expires_at"])
 
+        if user is None:  # if user not in cache
+            user = self.bot.get_user(user_id) or await self.bot.fetch_user(user_id)
```

Comment:
I think error handling would be needed for the API calls for two reasons:

1. `NotFound` is probably raised when a user deletes their account
2. It'd be silly to have the entire command fail because of an exception raised for a single non-critical API call. Who knows why `HTTPException` could be raised but it could happen...

Reply:
It doesn't look like discord.py will cache the user returned from the API call which isn't ideal. Could we cache it ourselves.

With the caching and error handling, perhaps it's becoming more effort than it's worth. But I think moderators, the ones going to use this command, should be the ones to comment on whether they'd be OK with not displaying names if not in the cache.

Reply:
I will add a exception catch to the `fetch_user` function. I don't change the display output for now, waiting for a mod opinion.

Reply:
@MarkKoz a ```py
try: 

except:``` Would be sufficient or should I use the error_handler?

Reply:
Yes, use a try-except to catch `discord.HttpException`.